include/jni/functions.hpp

```diff
@@ -589,7 +589,7 @@ namespace jni
     inline UniqueEnv AttachCurrentThread(JavaVM& vm)
        {
         JNIEnv* result;
-        CheckErrorCode(vm.AttachCurrentThread(&result, nullptr));
+        CheckErrorCode(vm.AttachCurrentThread(reinterpret_cast<void**>(&result), nullptr));
```

Comment:
@daedric  @jfirebaugh  FYI here is how I did it a while ago: https://github.com/rayglover-ibm/jni.hpp/commit/a29d5166dc9e3d55496f125f5149a7dbbd9a9749

(I'm also maintaining fixes for gcc 5.4, MSVC(2015/2017/C2), OpenJDK and OracleJDK for win/mac/ubuntu, and a CMake build for these platforms. Let me know if you're interested.)